docs/audio.rst

```diff
@@ -83,8 +86,215 @@ The built-in sounds can be called using ``audio.play(Sound.NAME)``.
+
+    .. py:attribute:: freq_start
+
+        Start Frequency in Hertz (Hz)
+
+    .. py:attribute:: freq_end
+
+        End Frequency in Hertz (Hz)
+
+    .. py:attribute:: duration
+
+        Duration of the sound (ms), eg: ``500``
+
+    .. py:attribute:: vol_start
+
+        Start volume value, range 0-255, eg: ``120``
+
+    .. py:attribute:: vol_end
+
+        End volume value, range 0-255, eg: ``255``
+
+    .. py:attribute:: wave
+
+        Type of wave shape, one of these values: ``WAVE_SINE``,
+        ``WAVE_SAWTOOTH``, ``WAVE_TRIANGLE``, ``WAVE_SQUARE``,
+        ``WAVE_NOISE`` (randomly generated noise).
+
+    .. py:attribute:: fx
+
+        Effect to add on the sound, one of the following values:
+        ``FX_TREMOLO``, ``FX_VIBRATO``, ``FX_WARBLE``, or ``None``.
+
+    .. py:attribute:: interpolation
+
+        The type of curve between the start and end
+        frequencies, different wave shapes have different rates of change
+        in frequency. One of the following values: ``INTER_LINEAR``,
+        ``INTER_CURVE``, ``INTER_LOG``.
+
+The arguments used to create any Sound Effect, including the built in effects,
+can be inspected by looking at each of the Effect instance attributes, or by
+converting the instance into a string (which can be done via ``str()``
+function, or by using a function that does the conversion automatically like
+``print()``).
+
+For example, with the :doc:`REPL </devguide/repl>` you can inspect the built
+in Effects::
+
+    >>> audio.Effect.CROAK
+    Effect(freq_start=..., freq_end=..., duration=..., vol_start=..., vol_end=..., wave=..., fx=..., interpolation=...)
+
+The built in Effects are immutable, so they cannot be changed. Trying to modify
+a built in Effect will throw an exception::
+
+    >>> audio.Effect.CLICK.duration = 1000
+    Traceback (most recent call last):
+        File "<stdin>", line 1, in <module>
+    TypeError: effect cannot be modified
+
+But a new one can be created like this::
+
+    >>> click_clone = Effect(audio.Effect.CLICK)
+    >>> click_clone.duration = 1000
+    >>>
+
+Built in Sound Effects
+----------------------
+
+Some pre-created Sound Effects are already available as examples. These can
+be played directly ``audio.play(audio.Effect.SQUEAK)``, or used as a base to
+create new effects ``audio.Effect(audio.Effect.SQUEAK, duration=2000)``.
+
+* ``audio.Effect.SQUEAK``
+* ``audio.Effect.WARBLE``
+* ``audio.Effect.CHIRP``
+* ``audio.Effect.CROAK``
+* ``audio.Effect.CLICK``
+
+Sound Effects Example
+---------------------
+
+::
+
+    from microbit import *
+
+    # Play a built in Sound Effect
+    audio.play(audio.Effect.CHIRP)
+
+    # Create a Sound Effect and immediately play it
+    audio.play(Effect(
+        freq_start=400,
+        freq_end=2000,
+        duration=500,
+        vol_start=100,
+        vol_end=255,
+        wave=audio.WAVE_TRIANGLE,
+        fx=audio.FX_VIBRATO,
+        interpolation=audio.LOG
+    ))
+
```

Comment:
I like this, it's nice and clear, but also think we should give some thought to when to document the positional, not keyword options.

@microbit-matt-hillsdon when autocomplete works well with positional arguments, it can be a really smooth editing experience - are there ways we could document the code here so that we get good positional argument help during autocomplete as long as the user hasn't used any keyword arguments 

(for example, if I start typing `audio.play(Effect(400` will the autocomplete in the editor be showing the parameter help from freq_start (as long as we re-order the parameters so start is first))